lib/pages/Mainpage.dart

```diff
@@ -0,0 +1,157 @@
+  final PageStorageBucket bucket = PageStorageBucket();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text("XPEHO"),
+        ),
+        body:PageStorage(bucket: bucket, child: currentScreen),
+        floatingActionButton: FloatingActionButton(
+          child:Icon(_centerIcons[currentTab]),
+          backgroundColor: Colors.orangeAccent,
+          onPressed: (){},
+        ),
+        floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
+        bottomNavigationBar: BottomAppBar(
+          shape: CircularNotchedRectangle(),
+          child:Container(
+            height: 60,
+            child:Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+              Row(
+                children:<Widget>[
+                  MaterialButton(
+                    key:Key("Cards"),
+                    minWidth:40,
+                    onPressed:(){
+                      setState((){
+                        currentScreen = LoyaltyCards();
+                        currentTab = 0;
+                      });
+                    },
+                    child:Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children:<Widget>[
+                        Icon(Icons.loyalty),
+                        Text(getString(context,'loyalty_cards'))
+                      ]
+                    )
+                  ),
+                  MaterialButton(
+                     key:Key("Lists"),
+                    minWidth:40,
+                    onPressed:(){
+                      setState((){
+                        currentScreen = Lists();
+                        currentTab = 1;
+                      });
+                    },
+                    child:Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children:<Widget>[
+                        Icon(Icons.list),
+                        Text(getString(context,'my_lists'))
+                      ]
+                    )
+                  ),
+                ]
+              ),
+              Expanded(
+                child:Padding(
+                  padding: const EdgeInsets.only(top:25.0),
+                  child:Center(
+                      child: Text(
+                        getString(context, _centerText[currentTab]),
+                        style: TextStyle(
+                          fontSize: 10.0,
+                        ),
+                      )
+                  ),
+                ),
+              ),
+              Row(
+                children:<Widget>[
+                  MaterialButton(
+                     key:Key("Share"),
+                    minWidth:40,
+                    onPressed:(){
+                      setState((){
+                        currentScreen = Share();
+                        currentTab = 2;
```

Comment:
Can you please create constants to call each page ? 0, 1, 2, 4 is kind of "magic numbers" and we don't understand what's the real signification.
Create constants like const SHARE_PAGE = 2;